orm_lib/inc/drogon/orm/BaseBuilder.h

```diff
@@ -0,0 +1,352 @@
+/**
+ *
+ *  @file BaseBuilder.h
+ *  @author Ken Matsui
+ *
+ *  Copyright 2022, Ken Matsui.  All rights reserved.
+ *  https://github.com/drogonframework/drogon
+ *  Use of this source code is governed by a MIT license
+ *  that can be found in the License file.
+ *
+ *  Drogon
+ *
+ */
+
+#pragma once
+
+#include <drogon/orm/DbClient.h>
+#include <drogon/utils/optional.h>
+#include <drogon/utils/string_view.h>
+#include <future>
+#include <memory>
+#include <string>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+#ifdef __cpp_impl_coroutine
+#include <drogon/utils/coroutine.h>
+#endif
+
+#ifdef _MSC_VER
+#define unreachable() __assume(false)
+#else
+#define unreachable() __builtin_unreachable()
+#endif
+
+namespace drogon
+{
+namespace orm
+{
+enum class Op
+{
+    /// =
+    eq,
+    /// !=
+    neq,
+    /// >
+    gt,
+    /// >=
+    gte,
+    /// <
+    lt,
+    /// <=
+    lte,
+    /// like
+    like,
+};
```

Comment:
Should we use the same type as:
https://github.com/drogonframework/drogon/blob/a1d2bd168b92ee6a0d38c4e7a83c383a288d4890/orm_lib/inc/drogon/orm/Criteria.h#L34-L48

Reply:
Fixed.